odl/solvers/nonsmooth/adupdates.py

```diff
@@ -0,0 +1,177 @@
+        is used.
+    callback : callable, optional
+        Function called with the current iterate after each iteration.
+    callback_loop : string, optional
+       If 'inner', the `callback` function is called after each inner
+       iteration, i.e., after each dual update. If 'outer', the `callback`
+       function is called after each outer iteration, i.e., after each primal
+       update.
+
+    Notes
+    -----
+    The algorithm as implemented here is described in the article [MF2015],
+    where it is applied to a tomography problem. It solves the problem
+
+    .. math::
+        \min_x \sum_{i=1}^m f_i(L_i x),
+
+    where :math:`f_i` are proper, convex and lower semicontinuous functions
+    and :math:`L_i` are linear, continuous operators for
+    :math:`i = 1, \ldots, m`. In an outer iteration, the solution is found
+    iteratively by an iteration
+
+    .. math::
+        x_{n+1} = \mathrm{arg\,min}_x \sum_{i=1}^m f_i(L_i x)
+            + \frac{\mu}{2} \|x - x_n\|^2
+
+    according to the proximal point algorithm. In the inner iteration, dual
+    variables are introduces for each of the components of the sum. The
+    Lagrangian of the problem is given by
+
+    .. math::
+        S_n(x; v_1, \ldots, v_m) = \sum_{i=1}^m (\langle v_i, L_i x \rangle
+            - f_i^*(v_i)) + \frac{\mu}{2} \|x - x_n\|^2.
+
+    Given the dual variables, the new primal variable :math:`\tilde x_{n+1}`
+    can be calculated by directly minimizing :math:`S_n` with respect to
+    :math:`x`. This corresponds to the formula
+
+    .. math::
+         \tilde x_{n+1} = x_n - \frac{1}{\mu} \sum_{i=1}^m L_i^* v_i.
+
+    The dual updates are executed according to the following rule:
+
+    .. math::
+        v_j^+ = \mathrm{Prox}_{\mu M_j^{-1}}
+            (v_j + \mu M_j^{-1} L_j \tilde x_{n+1}),
+
+    where :math:`M_j` is a diagonal matrix with positive diagonal entries such
+    that :math:`M_j - L_j L_j^*` is positive semidefinite. The matrix
+    :math:`M_j` is incorporated into this function via the variable `majs` in
+    one of the following ways:
+
+    * Setting `majs[i]` a positive float :math:`\gamma` corresponds to the
+    choice :math:`M_i = \gamma \mathrm{Id}`.
+
+    * Assume that `f_j` is a `SeparableSum`, then setting `majs[i]` a list
+    :math:`(\gamma_1, \ldots, \gamma_k)`of positive floats corresponds to the
+    choice of a block-diagonal matrix :math:`M_j`, where each block corresponds
+    to one of the space components and equals :math:`\gamma_i \mathrm{Id}`.
+
+    * Assume that `f_j` is a `L1_norm` or a `L2_norm_squared`, then setting
+    `majs[i]` a `f_j.domain.element` :math:`z` corresponds to the choice
+    :math:`\mathrm{diag}(z)`.
+
+    References
+    ----------
+    [MF2015] McGaffin, M G, and Fessler, J A. *Alternating dual updates
+    algorithm for X-ray CT reconstruction on the GPU*. IEEE Transactions
+    on Computational Imaging, 1.3 (2015), pp 186--199.
```

Comment:
Reference looks perfect. If you want to use a citation with link, which would be something for the short summary before `Parameters`, you can use a link like
```
`[MF2015] <URL for DOI, arXiv or whatever>`_
```
This is at least my current favorite way of doing it, because it provides a clickable link *and* a text reference in the `References` section.